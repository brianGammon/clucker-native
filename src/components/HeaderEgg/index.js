import HeaderEggComponent from './HeaderEgg';

export default HeaderEggComponent;
