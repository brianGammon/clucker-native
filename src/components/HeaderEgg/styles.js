import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  icon: {
    fontSize: 25,
    color: 'grey',
  },
  text: {
    color: 'grey',
    fontSize: 20,
    paddingRight: 10,
  },
});
