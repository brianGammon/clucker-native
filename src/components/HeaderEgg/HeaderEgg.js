import * as React from 'react';
import { Text } from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import styles from './styles';

type Props = {
  navigation: any,
};

export default ({ navigation }: { navigation: Props.navigation }) => {
  const chickenId = navigation.getParam('chickenId', null);
  const options = {};
  if (chickenId) {
    options.chickenId = chickenId;
    options.eggId = 'eggID1';
    options.date = '2018-10-19';
  }
  return (
    <Text
      onPress={() => navigation.navigate('EggModal', options)}
      style={styles.text}
    >
      +
      <Ionicons name="ios-egg" color="grey" style={styles.icon} />
    </Text>
  );
};
