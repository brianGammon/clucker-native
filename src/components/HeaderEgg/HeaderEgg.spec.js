import React from 'react';
import HeaderEgg from '.';

describe('HeaderEgg component:', () => {
  test('Should render', () => {
    const wrapper = shallow(<HeaderEgg />);
    expect(wrapper).toMatchSnapshot();
  });
});
