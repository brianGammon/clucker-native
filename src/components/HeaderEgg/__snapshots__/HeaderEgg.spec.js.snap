// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderEgg component: Should render 1`] = `
<Component
  onPress={[Function]}
  style={
    Object {
      "color": "grey",
      "fontSize": 20,
      "paddingRight": 10,
    }
  }
>
  +
  <Icon
    allowFontScaling={false}
    color="grey"
    name="ios-egg"
    size={12}
    style={
      Object {
        "color": "grey",
        "fontSize": 25,
      }
    }
  />
</Component>
`;
